client/modules/User/components/Collection.jsx

```diff
@@ -34,7 +29,7 @@ const Collection = ({ collectionId }) => {
 
   const isOwner = () =>
     user != null &&
-    user.username &&
+    typeof user.username !== 'undefined' &&
     collection?.owner?.username === user.username;
```

Comment:
IMO it's important to use descriptive variable names like `currentUsername` and `ownerUsername` to avoid bugs like the one that you fixed here.

I know that I can be rude sometimes but I think you can understand my frustrations.  Because I already converted this component over a year ago and it was cleanly written and did not have that sort of mistake.

Reply:
Thanks for catching these, there's probably still a lot of other areas this component could be improved here! 

Since I wasn't able to get through all of your work and made some changes to these files that differed from yours, I do get your frustrations on how it probably feels like this could've been preventable. Although it'll be a lot of work since I didn't keep up with the PRs from back then, if you'd like to, we can get still try to get them in. I could be wrong, but I think most of them could probably be merged with some minimal changes except for a few (like the Common Table Components and the CodeMirror one, esp since it's part of the fellowship). I was also planning to [upgrade the Node.js minor version](https://github.com/processing/p5.js-web-editor/compare/develop...update-node-version) in the near future, which I think might conflict with [this one](https://github.com/processing/p5.js-web-editor/pull/2220). 

This is a bit of a tangent and definitely depends on the context, but I also personally feel that it's okay for code to end up a little messy and have mistakes sometimes! While I do think it's important to try not to make them, I don't want to view them negatively because I feel like they're a really normal part of any process, and in most cases, can usually be fixed.

Reply:
Wanted to add that I meant the latter more as a point of conversation, rather than disagreeing!